src/main/java/bio/terra/service/filedata/google/firestore/FireStoreDao.java

```diff
@@ -388,16 +459,31 @@ FireStoreDirectoryEntry computeDirectory(Firestore firestore,
             .checksumCrc32c(crc32cChecksum)
             .checksumMd5(md5Checksum)
             .size(totalSize);
-        directoryDao.updateDirectoryEntry(firestore, snapshotId, dirEntry);
+        updateEntry(firestore, snapshotId, dirEntry, updateBatch);
 
         return dirEntry;
     }
 
+    private void updateEntry(
+        Firestore firestore,
+        String snapshotId,
+        FireStoreDirectoryEntry entry,
+        List<FireStoreDirectoryEntry> updateBatch) throws InterruptedException {
+
+        updateBatch.add(entry);
+        int batchSize = configurationService.getParameterValue(FIRESTORE_SNAPSHOT_BATCH_SIZE);
+        if (updateBatch.size() >= batchSize) {
+            logger.info("Snapshot compute dating batch of {} directory entries", batchSize);
```

Comment:
What is a dating batch?

Reply:
I have no idea. Don't you just love voice code dictation?